core/lib/spree/core/validators/email.rb

```diff
@@ -13,11 +13,16 @@ module Spree
   #
   class EmailValidator < ActiveModel::EachValidator
     EMAIL_REGEXP = URI::MailTo::EMAIL_REGEXP
+    EMAIL_DOMAIN_REGEXP = /\.[a-z]+/i
 
     def validate_each(record, attribute, value)
-      unless EMAIL_REGEXP.match? value
+      unless valid_email?(value)
         record.errors.add(attribute, :invalid, { value: value }.merge!(options))
       end
     end
+
+    def valid_email?(value)
+      EMAIL_REGEXP.match?(value) && EMAIL_DOMAIN_REGEXP.match?(value)
```

Comment:
AFAIK the key issue is that `URI::MailTo::EMAIL_REGEXP` doesn't validate that addresses end with a domain suffix. Perhaps you can maintain the protection of the original regex and just add an additional validation that the address includes a domain suffix?

Reply:
I'd prefer using a single step for the validation, it's just a matter to find a good and battle testes regexp :slightly_smiling_face: